core/src/test/scala/org/apache/spark/util/collection/ExternalAppendOnlyMapSuite.scala

```diff
@@ -550,12 +550,11 @@ class ExternalAppendOnlyMapSuite extends SparkFunSuite
 
   test("force to spill for external aggregation") {
     val conf = createSparkConf(loadDefaults = false)
-      .set("spark.shuffle.memoryFraction", "0.01")
-      .set("spark.memory.useLegacyMode", "true")
-      .set("spark.testing.memory", "100000000")
+      .set("spark.memory.storageFraction", "0.999")
+      .set("spark.testing.memory", "471859200")
```

Comment:
This is the minimum allowed value. The new setting 0.999 mimics having almost no shuffle memory